packages/main/src/components/AnalyticalTable/pluginHooks/useRowDisableSelection.tsx

```diff
@@ -0,0 +1,72 @@
+import { CheckBox } from '@ui5/webcomponents-react/lib/CheckBox';
+import { TableSelectionMode } from '@ui5/webcomponents-react/lib/TableSelectionMode';
+import React from 'react';
+import { useStyles } from '../index';
+
+type DisableRowSelectionType = string | ((row: Record<any, any>) => boolean);
+
+/**
+ * A plugin hook for disabling row selection of specific rows.
+ * __Note:__ The "Select All" checkbox is not available with this hook.
+ *
+ * @param disableRowSelection - Can be either a `string` or a `function`. `string:` Defines the key in the dataset for disabling rows. If the value of the key is `true`, then the row will not be selectable. `function:` Programmatically disable rows for selection. The function receives the current row as parameter.
+ */
+export const useRowDisableSelection = (disableRowSelection: DisableRowSelectionType) => {
+  const classes = useStyles();
+  const getRowProps = (rowProps, instance) => {
+    const { row } = instance;
+    if (typeof disableRowSelection === 'function' && disableRowSelection?.(row) === true) {
+      row.disableSelect = true;
+      return { ...rowProps, onClick: undefined, className: classes.tr };
+    }
+    if (typeof disableRowSelection === 'string' && row.original[disableRowSelection] === true) {
```

Comment:
can we reuse the accessor logic from the table? a deep accessor like `fried.age` wouldn't work in that case